mpt_api_client/http/resource.py

```diff
@@ -111,9 +136,8 @@ def save(self) -> Self:
             contact.save()
 
         """
-        if not self.resource_:
-            raise ValueError("Unable to save resource that has not been set.")
-        self.update(self.resource_.to_dict())
+        self._assert_resource_is_set()
+        self.update(self.resource_.to_dict())  # type: ignore[union-attr]
```

Comment:
Take care with this ignore, The error comes from the typing in the init method, self.resource can be None, so at some point we'll get an error because --> None.to_dict()

We have two options here, either check that self.resource is not None before the update, or change the typing to disallow None.